### DISCUSSION TITLE
Question about the outline view(secondary sidebar)? Change color and icons

### DISCUSSION THREAD

**Discussion by JWigh** [codefori/vscode-ibmi#2254] [General]
Is it possible to change the Coloring, the icons, and the size of everything(see picture)

When you coming from RDi with the outline this can be a little bit hard to read and see because of the coloring and the icons doesnt pop out.

I have fiddle with coloring but i would also like the icons to be changed

 
![markup](https://github.com/user-attachments/assets/8f32881e-a893-43eb-83a7-1ddf95d53cd6)

**Comment by chrjorgensen** [comment]
@JWigh You may try switching the product icons: Press `F1` and type `product icon` and you will get this prompt, where you can change the icon theme or install a new icon theme.

![billede](https://github.com/user-attachments/assets/d114f073-dc9f-4f71-b7cc-f638273c2870)

**Reply by JWigh** [reply]
Ok i can see some difference but i would like the old style RDi or Visual studio icon birghter and full of color

**Reply by chrjorgensen** [reply]
The colours and icons are controlled by VS Code, not by this extension - we only specify what icon (by name) we want and what text should appear, and the themes control the rest. So unfortunately we can't change it to your liking...

**Reply by JWigh** [reply]
It the Icons specified in a JSON file or Specified in Typescript code and if so where is that code situated in the repository?

**Reply by sebjulliand** [reply]
The extension provides a list of symbols to the outline view. No icons or colors are specified.
Then the outline view display the symbols and pick the icon and color based on their type. I guess the only way for you to have the icons and colors you want is to define [your own theme](https://code.visualstudio.com/api/extension-guides/file-icon-theme) for the outline view.

**Reply by JWigh** [reply]
Ok, thats probably the best solution, but I would like to know where in extension code is mapping between code and symbols taken place?

**Reply by sebjulliand** [reply]
This must be it: https://github.com/codefori/vscode-rpgle/blob/6c335cb60a826e775834506407c001fb0372a9ea/extension/server/src/providers/documentSymbols.ts

**Reply by JWigh** [reply]
I Probably have to give in and keep on using the RDi tool.

**Reply by JWigh** [reply]
A question to everybody? How hard would it to create an extension that behave like the outline view in rdi.  as  it doesn’t exist code for ibm i  like using PDm .

**Reply by thomprl** [reply]
> A question to everybody? How hard would it to create an extension that behave like the outline view in rdi. as it doesn’t exist code for ibm i like using PDm .

Marcus Litters is actually working on a RDi outline clone for VSCode.  But I'm not sure when it will be ready.

**Reply by JWigh** [reply]
Thanks I couldn’t start understanding because jobb computer is locked so I can’t fetch npm packages
